lib/src/util.dart

```diff
@@ -423,6 +423,22 @@ Iterable<AstNode> allDescendants(AstNode node) sync* {
 Iterable<T> allDescendantsOfType<T extends AstNode>(AstNode node) =>
     allDescendants(node).whereType<T>();
 
+/// Finds a dependency within a pubspec and returns the [VersionRange].
+///
+/// NOTE: This logic is not very flexible and doesn't work for hosted dependencies.
+VersionRange? getDependencyVersion(String pubspecContent, String dependency) {
```

Comment:
>doesn't work for hosted dependencies.

Perhaps we could capture that in the name so that it's more obvious? e.g., `getNonHostedDependencyVersion`

Reply:
Also, if it's not very flexible, but is sufficient for this use-case, perhaps it should be a private (or `@visibleForTesting`) function within the file it's used, so that it doesn't get used naively in other code in the future.